hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1390,6 +1390,10 @@ public boolean shouldAllowMultiWriteOnSameInstant() {
     return getBoolean(ALLOW_MULTI_WRITE_ON_SAME_INSTANT_ENABLE);
   }
 
+  public boolean shouldDropPartitionColumns() {
```

Comment:
why is this needed? Isn't there aleady a getter for this config?

Reply:
Did not find any in `HoodieWriteConfig`. The usages in other places depends on `DataSourceWriteOptions` which was warning about adding a circular dependency.